### ISSUE TITLE
SWF file support

### ISSUE THREAD

**Issue by lvbnhbtd** [pietie/HyperSearch#27]
Hello dear pietie. Most likely there will be no answer. But I'll try. 
Can add support for SWF files to HyperSearch? HyperSpin can be used in wheel swf, but HyperSearch cannot see them.
p.s. excuse me for my English.

**Comment by pietie** [comment]
Hi

I don't recall "wheel swf". How does it work and where does the data you want to search come from if not from the Xml databases? If it's in the actual SWF then that might be challenging 

Edit: 
Oh after a quick google search I think I understand now. It is the wheel images that are swf? Can you send me some? HyperSearch uses WPF which does not support SWF out of the box. There are some ways but it might not work that great. If you can send me some samples I'll play around with it but don't get your hopes up :)

**Comment by lvbnhbtd** [comment]
Yes, maybe use swfWheel in HyperSpin. This conversion is necessary for the Wheel to display correctly in 16x9 mode. 
Wheel - https://drive.google.com/drive/folders/1xCQupX4gPCUNy-lmOYeZr2R1oo0ME_5t?usp=sharing
Example of work - https://www.youtube.com/watch?v=Xq7c5il-kFY

**Comment by pietie** [comment]
Hi

So it's not looking that great. I have to convert the SWF to XAML which is not always a perfect conversion.  
You can try the latest build at [Releases](https://github.com/pietie/HyperSearch/releases)
I think the animation timing is "out". The SWFs also all seem to have some arbitrary offset during the animation. I try and pull this offset from the animation automatically but if I don't start the animation immediately the images are not in the correct place.

You might want to setup a test copy of your Hyperspin folders. It does not modify anything but it does generate a ton of new files in the conversion process. This is a once off process so when you search the UI might lock up for a while if your search is too wide. So try searching for something more specific first.

You should see a new .xaml file with  a Resources folder generated for each swf you have.

Have a look and see how well it works for you. I find that if there are two many items loaded at the same time then performance suffers greatly which makes this solution unusable at the moment.

![2020-11-30_11-52-15](https://user-images.githubusercontent.com/5171545/100595742-ff88b580-3303-11eb-8a3c-287df010e7c9.gif)

**Comment by lvbnhbtd** [comment]
Thanks you!
